### PR TITLE
Lazy loading engine CSS

### PR DESCRIPTION
**Pull request by thoov**
This builds off of https://github.com/embroider-build/embroider/pull/434 and must land after that one

- [x] Update tests
- [ ] #434 has landed

### CODE REVIEW

File: packages/core/src/app.ts [hunk 1]
```diff
@@ -846,7 +848,8 @@ export class AppBuilder<TreeNames> {
     engine: Engine,
     childEngines: Engine[],
     prepared: Map<string, InternalAsset>,
-    entryParams?: Partial<Parameters<typeof entryTemplate>[0]>
+    entryParams?: Partial<Parameters<typeof entryTemplate>[0]>,
+    lazyStyles?: string[]
```
Comment (thoov) [inline packages/core/src/app.ts:852]: We should probably rename this to something more generic

**Comment by GCheung55** [comment]
@thoov Just following up on this issue since #434 was merged.

**Comment by thoov** [comment]
@GCheung55 The underlining issue still remains and @ef4 and I have had discussions around if there are better ways of fixing this. For the internal project we are working on it did not actually need this fix so for prioritization reasons we paused work on this. My plan was in the next few weeks / month pick this back up and also address some other bugs related to engines.

**Comment by GCheung55** [comment]
@thoov whats the reason for closing this PR?

**Comment by GCheung55** [comment]
Nm. I see it’s been superseded by #591

**Comment by thoov** [comment]
@GCheung55 Yes, I closed this in favor of #591. Sorry I didn't make that more clear

**Pull request closed without merging**